include/hlslib/common/OpenCL.h

```diff
@@ -226,23 +245,44 @@ cl::CommandQueue CreateCommandQueue(cl::Context const &context,
   return commandQueue;
 }
 
+#ifdef HLSLIB_XILINX
+cl_mem_flags BankToFlag(MemoryBank memoryBank, bool failIfUnspecified,
+                        XilinxDDRBankFlags &refBankFlags) {
+  switch (memoryBank) {
+  case MemoryBank::bank0:
+    return refBankFlags.kMemoryBank0();
+  case MemoryBank::bank1:
+    return refBankFlags.kMemoryBank1();
+  case MemoryBank::bank2:
+    return refBankFlags.kMemoryBank2();
+  case MemoryBank::bank3:
+    return refBankFlags.kMemoryBank3();
+  case MemoryBank::unspecified:
+    if (failIfUnspecified) {
+      ThrowRuntimeError("Memory bank must be specified.");
+    }
+  }
+  return 0;
+}
+#else
 cl_mem_flags BankToFlag(MemoryBank memoryBank, bool failIfUnspecified) {
   switch (memoryBank) {
-    case MemoryBank::bank0:
-      return kMemoryBank0;
-    case MemoryBank::bank1:
-      return kMemoryBank1;
-    case MemoryBank::bank2:
-      return kMemoryBank2;
-    case MemoryBank::bank3:
-      return kMemoryBank3;
-    case MemoryBank::unspecified:
-      if (failIfUnspecified) {
-        ThrowRuntimeError("Memory bank must be specified.");
-      }
+  case MemoryBank::bank0:
+    return kMemoryBank0;
+  case MemoryBank::bank1:
+    return kMemoryBank1;
+  case MemoryBank::bank2:
+    return kMemoryBank2;
+  case MemoryBank::bank3:
+    return kMemoryBank3;
+  case MemoryBank::unspecified:
+    if (failIfUnspecified) {
+      ThrowRuntimeError("Memory bank must be specified.");
+    }
```

Comment:
We shouldn't need separate functions for Xilinx and Intel here. If it's really necessary, put them in `xilinx/OpenCL.h` and `intel_fpga/OpenCL.h`